groovy/gradle.java/src/org/netbeans/modules/gradle/java/classpath/AbstractGradleClassPathImpl.java

```diff
@@ -132,8 +128,15 @@ static void addAllFile(Collection<URL> ret, @NonNull Collection<File> files) {
             if (fo != null) {
                 if (FileUtil.isArchiveFile(fo)) {
                     ret.add(FileUtil.getArchiveRoot(fo).toURL());
-                } else {
-                    ret.add(fo.toURL());
+                } if (fo.isFolder()) {
+                    String surl = fo.toURL().toString();
+                    if (!surl.endsWith("/")) {
+                        try {
+                            ret.add(new URL(surl + "/")); //NOI18N
```

Comment:
Not sure if this is necessary - if fo is a folder, then its toURL() should end with a slash? Except for potentially a rare race condition, where the file ceases to exist between the fo.isFolder() and fo.toURL(). Anyway, if you want to keep that, shouldn't hurt.